xwayland/xwm.c

```diff
@@ -935,15 +936,25 @@ static void xwm_handle_configure_notify(struct wlr_xwm *xwm,
 		return;
 	}
 
-	xsurface->x = ev->x;
-	xsurface->y = ev->y;
-	xsurface->width = ev->width;
-	xsurface->height = ev->height;
+	bool geometry_changed =
+		(xsurface->x != ev->x || xsurface->y != ev->y ||
+		 xsurface->width != ev->width || xsurface->height || ev->height);
+
+	if (geometry_changed) {
+		xsurface->x = ev->x;
+		xsurface->y = ev->y;
+		xsurface->width = ev->width;
+		xsurface->height = ev->height;
+	}
 
 	if (xsurface->override_redirect != ev->override_redirect) {
 		xsurface->override_redirect = ev->override_redirect;
 		wlr_signal_emit_safe(&xsurface->events.set_override_redirect, xsurface);
 	}
+
+	if (geometry_changed) {
+		wlr_signal_emit_safe(&xsurface->events.set_geometry, NULL);
```

Comment:
Since this is only about geometry, I'm fine with emitting this before `set_override_redirect` if you want to change that back.

Reply:
I don't think the order matters much.

Reply:
Agreed.